spec/ruby/core/regexp/match_spec.rb

```diff
@@ -91,15 +91,6 @@
     end
   end
 
-  it "resets $~ if passed nil" do
-    # set $~
-    /./.match("a")
-    $~.should be_kind_of(MatchData)
-
-    /1/.match(nil)
-    $~.should be_nil
-  end
```

Comment:
It would be better to add a `ruby_version_is ""..."2.6"` guard instead of removing, since the behavior continues to apply to older versions.
Could you also add specs for the new behavior?

Reply:
Addresses in https://github.com/ruby/ruby/pull/1506/commits/1041650de2dd73e4f22271a5baf48195ecf7827c ~